working/1426-extension-types/feature-specification.md

```diff
@@ -888,15 +895,28 @@ extension type, e.g., `myList is List<nat>`.*
 We could use the following mechanism to enable casts to a protected
 extension type to succeed at run time:
 
-Assume that `E` is a protected extension type that declares a `bool get
-verifyThis` getter.
+Assume that `E` is a protected extension type that declares a
+`bool get verifyThis` getter.
 
 If such a getter exists, then the execution of a type cast `c` of the form
 `o as E` proceeds as follows: First, a cast `o as T` is executed, where `T`
 is the instantiated on-type corresponding to `E`. If this cast succeeds
-then `o.verifyThis` is evaluated to an object `o1`. If `o1` is the true
-object then `c` completes normally and yields `o`; otherwise `c` encounters
-a dynamic type error.
+then `this` is bound to `o` and `verifyThis` is evaluated to an object
+`o1`. If `o1` is the true object then `c` completes normally and yields
+`o`; otherwise `c` encounters a dynamic type error.
```

Comment:
Can you do `o is E` too? (which can still throw since you're running user code).

Reply:
That would definitely be the intention, and I don't see why it would be technically harder than supporting `o as E`.